### PR TITLE
Add script to create mapping files during build

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for https://github.com/Azure/azure-powershell/issues/6232 and https://github.com/Azure/azure-powershell/issues/6233

Add script to create three mapping files during build for the following tasks:
- building solutions
- generating help and running Static Analysis
- running tests

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
Link to PR with sample run times per service: https://github.com/Azure/azure-powershell/pull/6730

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/925/

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/952/

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServiceTests.cs [hunk 1]
```diff
@@ -30,35 +30,35 @@ public WebServiceTests(ITestOutputHelper output)
             XunitTracingInterceptor.AddToContext(this.interceptor);
         }
 
-        [Fact]
+        [Fact(Skip = "MachineLearning tests need to be re-enabled, as outlined in issue https://github.com/Azure/azure-powershell/issues/6684")]
```
Comment (markcowl) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServiceTests.cs]: why are all of these skips included?

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/BaseTestController.cs [hunk 2]
```diff
@@ -152,10 +152,10 @@ private void SetupManagementClients(MockContext context)
             HttpClientHelperFactory.Instance = new TestHttpClientHelperFactory(credentials);
 
             helper.SetupManagementClients(
-                this.resourceManagementClient, 
-                subscriptionClient, 
-                this.serviceClient, 
-                authManagementClient, 
+                this.resourceManagementClient,
```
Comment (markcowl) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/BaseTestController.cs]: why are these changes included?

File: src/ResourceManager/ManagementPartner/Commands.Partner.Test/ScenarioTests/ManagementPartnerTests.cs [hunk 3]
```diff
@@ -31,35 +31,35 @@ public ManagementPartnerTests(Xunit.Abstractions.ITestOutputHelper output)
             XunitTracingInterceptor.AddToContext(_logger);
         }
 
-        [Fact]
+        [Fact(Skip = "ManagementPartner tests need to be re-enabled, as outlined in issue https://github.com/Azure/azure-powershell/issues/6678")]
```
Comment (markcowl) [inline src/ResourceManager/ManagementPartner/Commands.Partner.Test/ScenarioTests/ManagementPartnerTests.cs]: ??

File: src/ResourceManager/Maps/Maps.Test/Commands.Maps.Test.csproj [hunk 4]
```diff
@@ -155,13 +159,29 @@
...
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Maps.Test.ScenarioTests.MapsAccountTests\TestGetAccountKeys.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Maps.Test.ScenarioTests.MapsAccountTests\TestGetAccounts.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Maps.Test.ScenarioTests.MapsAccountTests\TestNewAccount.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (markcowl) [inline src/ResourceManager/Maps/Maps.Test/Commands.Maps.Test.csproj]: ??

File: src/ResourceManager/Maps/Maps.Test/TestController.cs [hunk 5]
```diff
@@ -30,6 +29,7 @@
 using TestBase = Microsoft.Azure.Test.TestBase;
 using TestUtilities = Microsoft.Azure.Test.TestUtilities;
 using LegacyResourceManagementClient = Microsoft.Azure.Management.ResourceManager.ResourceManagementClient;
+using Microsoft.Azure.Management.Internal.Resources;
```
Comment (markcowl) [inline src/ResourceManager/Maps/Maps.Test/TestController.cs]: ??

File: src/ResourceManager/Maps/Maps.sln [hunk 6]
```diff
@@ -22,6 +22,12 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Maps.Test", "Maps.
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Common.Authentication.ResourceManager", "..\Common\Commands.Common.Authentication.ResourceManager\Commands.Common.Authentication.ResourceManager.csproj", "{69C2EB6B-CD63-480A-89A0-C489706E9299}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Resources", "..\Resources\Commands.Resources\Commands.Resources.csproj", "{E1F5201D-6067-430E-B303-4E367652991B}"
```
Comment (markcowl) [inline src/ResourceManager/Maps/Maps.sln]: ??

File: src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/MediaTests.ps1 [hunk 7]
```diff
@@ -115,7 +115,7 @@ function Test-Media
 
   <# Create a media service with multiple storage accounts #>
   $tags = @{"tag1" = "value1"; "tag2" = "value2"}
-  $mediaService = New-AzureRmMediaService -ResourceGroupName $rgname -AccountName $accountName -Location $location -StorageAccounts $storageAccounts -Tags $tags
+  $mediaService = New-AzureRmMediaService -ResourceGroupName $rgname -AccountName $accountName -Location $location -StorageAccounts $storageAccounts -Tag $tags
```
Comment (markcowl) [inline src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/MediaTests.ps1]: why is this change in this PR?

File: src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/TestController.cs [hunk 8]
```diff
@@ -70,7 +70,7 @@ protected TestController()
 
         protected void SetupManagementClients(MockContext context)
         {
-            ResourceManagementClient = TestBase.GetServiceClient<ResourceManagementClient>(_csmTestFactory);
+            ResourceManagementClient = context.GetServiceClient<ResourceManagementClient>(TestEnvironmentFactory.GetTestEnvironment());
```
Comment (markcowl) [inline src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/TestController.cs]: ??

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Test/UnitTests/Cmdlets/IaaS/Extensions/Diagnostics/DiagnosticsHelperTest.cs [hunk 9]
```diff
@@ -11,7 +11,7 @@ public class DiagnosticsHelperTest
             return init.Replace(System.Environment.NewLine, "").Replace(" ", "");
         }
 
-        [Fact]
+        [Fact(Skip = "Investigate failure with JSON formatting.")]
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Test/UnitTests/Cmdlets/IaaS/Extensions/Diagnostics/DiagnosticsHelperTest.cs]: why is this part of this PR?

File: tools/CreateFilterMappings.ps1 [hunk 10]
```diff
@@ -0,0 +1,457 @@
...
+        [Parameter(Mandatory = $true)]
+        [hashtable]$Mappings,
+
+        [Parameter(Mandatory = $true)]
+        [string]$SolutionPath
+    )
+
+    $ProjectDependencies = @()
+    $Content = Get-Content -Path $SolutionPath
+    $Content | Select-String -Pattern "`"[a-zA-Z.]*`"" | % { $_.Matches[0].Value.Trim('"') } | where { $_ -notlike "Test*" -and $_ -notlike "*.Test*" -and $_ -notlike "*Common*" } | % { $ProjectDependencies += $_ }
```
Comment (markcowl) [inline tools/CreateFilterMappings.ps1]: So, don't we want to specifically match ```<HintPath>path</HintPath>```, I worry that this match is a bit overly broad.

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServiceTests.cs (hunk 1 above)
Comment (cormacpayne) [reply src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServiceTests.cs]: None of the MachineLearning tests are currently being run as a part of CI, and when they are enabled with the new `TestMappings.json` file generated by the script, a few of the tests fail and cannot be immediately resolved. I opened the issue on the MachineLearning team to take a look and resolve the failures.

File: src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/BaseTestController.cs (hunk 2 above)
Comment (cormacpayne) [reply src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/BaseTestController.cs]: I have an extension in VS that trims trailing whitespace (😀), so outside of the change in instantiating the ResourceManagementClient, everything else in this file is whitespace trimming

File: src/ResourceManager/ManagementPartner/Commands.Partner.Test/ScenarioTests/ManagementPartnerTests.cs (hunk 3 above)
Comment (cormacpayne) [reply src/ResourceManager/ManagementPartner/Commands.Partner.Test/ScenarioTests/ManagementPartnerTests.cs]: Similar to MachineLearning, these tests weren't being run during the build and I opened an issue on the ManagementPartner team to take a look. It appears that these tests have been fixed with https://github.com/Azure/azure-powershell/pull/6817, so I will update the PR to reflect this

File: src/ResourceManager/Maps/Maps.Test/Commands.Maps.Test.csproj (hunk 4 above)
Comment (cormacpayne) [reply src/ResourceManager/Maps/Maps.Test/Commands.Maps.Test.csproj]: Similar to MachineLearning, the Maps tests weren't being run during the build, but I was able to make a few changes to get them to run successfully on the build. This included:
- Adding their `.json` recordings to their test project
- Updating the ResourceManagementClient instantiated in their test controller
- Making sure the Resources project was built with the Maps solution

File: src/ResourceManager/Maps/Maps.Test/TestController.cs (hunk 5 above)
Comment (cormacpayne) [reply src/ResourceManager/Maps/Maps.Test/TestController.cs]: See comment above

File: src/ResourceManager/Maps/Maps.sln (hunk 6 above)
Comment (cormacpayne) [reply src/ResourceManager/Maps/Maps.sln]: See comment above

File: src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/MediaTests.ps1 (hunk 7 above)
Comment (cormacpayne) [reply src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/MediaTests.ps1]: Similar to MachineLearning, the Media tests weren't being run during the build, but I was able to make a few changes to get them to run successfully on the build. This included:
- Changing the `-Tags` parameter to `-Tag`
- Changing the type of ResourceManagementClient instantiated in the test controller

File: src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/TestController.cs (hunk 8 above)
Comment (cormacpayne) [reply src/ResourceManager/Media/Commands.Media.Test/ScenarioTests/TestController.cs]: See comment above

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Test/UnitTests/Cmdlets/IaaS/Extensions/Diagnostics/DiagnosticsHelperTest.cs (hunk 9 above)
Comment (cormacpayne) [reply src/ServiceManagement/Compute/Commands.ServiceManagement.Test/UnitTests/Cmdlets/IaaS/Extensions/Diagnostics/DiagnosticsHelperTest.cs]: I will need to double-check, but I don't think these RDFE tests weren't being run as a part of the build, and some of them were consistently failing when run on Jenkins.

File: tools/CreateFilterMappings.ps1 (hunk 10 above)
Comment (cormacpayne) [reply tools/CreateFilterMappings.ps1]: This function matches projects (excluding test and common projects) built in a provided solution.

For example it will match `Commands.Compute` from the line 

```
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commands.Compute", "Commands.Compute\Commands.Compute.csproj", "{52643BD5-6378-49BD-9F6E-DAC9DD8A867B}"
```

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1015/

**Comment by cormacpayne** [comment]
@markcowl @maddieclayton opened an issue on the Compute.ManagedService contact to take a look at the failing test that was enabled as a part of this PR.

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/1019/

**Comment by markcowl** [comment]
@cormacpayne Dude, you are terrible at being on vacation
